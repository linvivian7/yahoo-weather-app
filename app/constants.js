export const LOCATION_FORM_NAME = 'locationForm';
