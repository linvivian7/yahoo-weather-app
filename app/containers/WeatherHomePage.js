import React from 'react';

import Nav from '../components/nav';
import CurrentStatusCard from '../components/currentStatusCard';
import SunStatusCard from '../components/SunStatusCard';
import WindStatusCard from '../components/WindStatusCard';

export default class WeatherHomePage extends React.PureComponent {

    componentDidMount() {
        const {
            onSubmit,
            searchTerm
        } = this.props;

        onSubmit({location: searchTerm});
    }

    onHomePageLinkClick(e) {
        e.preventDefault();

        this.props.onHomePageLinkClick();
    }

    onForecastLinkClick(e) {
        e.preventDefault();

        this.props.onForecastLinkClick();
    }

    render() {
        const {
            searchTerm,
            onChange,
            onSubmit,
            timezone,
            weatherInfo
        } = this.props;
        let currentStatusCard;
        let sunStatusCard;
        let windStatusCard;

        if (weatherInfo) {
            const {
                astronomy,
                atmosphere,
                item,
                units,
                wind
            } = weatherInfo;

            currentStatusCard = (
                <CurrentStatusCard
                    atmosphere= { atmosphere }
                    condition={ item.condition }
                    searchTerm={ searchTerm }
                />);
            sunStatusCard = (<SunStatusCard timezone={ timezone } astronomy={ astronomy } />);
            windStatusCard = (<WindStatusCard timezone={ timezone } wind={ wind } units={ units } />);
        }

        return (
            <div>
                <Nav
                    location= { searchTerm }
                    onChange={ onChange }
                    onSubmit={ onSubmit }
                    onHomePageLinkClick={ this.onHomePageLinkClick.bind(this) }
                    onForecastLinkClick={ this.onForecastLinkClick.bind(this) }
                />
                <div className="page-container">
                    { currentStatusCard }
                    { sunStatusCard }
                    { windStatusCard }
                </div>
            </div>
        );
    }
}
