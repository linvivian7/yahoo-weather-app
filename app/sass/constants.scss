@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

$break-width: 768px;
$temp-hover-color: #2196F3;

$text-color-sandstone: #F4CC70;
$text-color-burnt-orange: #DE7A22;
$text-color-sea: #20948B;
$text-color-lagoon: #6AB187;

$font-family-primary: 'Roboto Condensed', sans-serif;
$font-family-nav: 'Open Sans', sans-serif;

$font-size-small: 16px;
$font-size-medium: 24px;
$font-size-med-large: 60px;
$font-size-large: 96px;
$font-size-x-large: 112px;

$icon-size-x-small: 32px;
$icon-size-small: 48px;
$icon-size-medium: 60px;
$icon-size-x-large: 120px;

$padding-x-small: 4px;
$padding-small: 16px;
$padding-medium: 24px;
$padding-med-large: 36px;
$padding-large: 48px;

$margin-x-small: 4px;
$margin-small: 16px;
$margin-medium: 24px;
$margin-med-large: 36px;
$margin-large: 48px;

@mixin status-card {
    box-shadow: 0 2.5px 8px 0 rgba(0,0,0,.2);
    color: white;
    display: grid;
    font-family: $font-family-primary;
    margin-bottom: 15px 0;
    align-items: center;
}
