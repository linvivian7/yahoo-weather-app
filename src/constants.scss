@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

$break-width: 768px;

$color-grey: #AAA;
$color-sunrise: #EAECC6;
$color-sandstone: #F4CC70;
$color-burnt-orange: #DE7A22;
$color-lagoon: #6AB187;
$color-sea: #20948B;

$font-family-primary: 'Roboto Condensed', sans-serif;
$font-family-nav: 'Open Sans', sans-serif;

$font-size-x-small: 12px;
$font-size-small: 3vw;
$font-size-medium: 4vw;
$font-size-med-large: 64px;
$font-size-large: 96px;
$font-size-x-large: 112px;

$icon-size-x-small: 32px;
$icon-size-small: 48px;
$icon-size-medium: 60px;
$icon-size-x-large: 120px;

$padding-x-small: 4px;
$padding-small: 16px;
$padding-medium: 24px;
$padding-med-large: 36px;
$padding-large: 48px;

$margin-x-small: 4px;
$margin-small: 16px;
$margin-medium: 24px;
$margin-med-large: 36px;
$margin-large: 48px;

@mixin status-card {
    box-shadow: 0 2.5px 8px 0 rgba(0,0,0,.2);
    color: white;
    display: grid;
    font-family: $font-family-primary;
    margin-bottom: 15px 0;
    align-items: center;

    @media (min-width: $break-width) {
        padding: 1em;
    }
}
