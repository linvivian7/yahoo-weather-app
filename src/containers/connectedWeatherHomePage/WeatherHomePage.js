import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import Menu from '../../components/menu';
import CurrentStatusCard from '../../components/currentStatusCard';
import SunStatusCard from '../../components/sunStatusCard';
import WindStatusCard from '../../components/windStatusCard';
import { GITHUB_LINK, YAHOO_WEATHER_LINK } from '../../constants';

import './weatherHomePage.scss';

export default class WeatherHomePage extends React.PureComponent {
    constructor(props) {
        super(props);

        const isLoading = props.weatherInfo ? false : true;

        this.state = { isLoading };
    }

    componentDidMount() {
        const {
            onSubmit,
            searchTerm,
            weatherInfo
        } = this.props;

        if (!weatherInfo) {
            onSubmit({ location: searchTerm });
        }
    }

    componentWillReceiveProps({ isLoading }) {
        this.setState({ isLoading });
    }

    render() {
        const {
            searchTerm,
            onChange,
            onSubmit,
            onHomePageLinkClick,
            onForecastLinkClick,
            onUnitToggleChange,
            timezone,
            weatherInfo
        } = this.props;
        let { isLoading } = this.state;
        let currentStatusCard;
        let sunStatusCard;
        let windStatusCard;

        if (weatherInfo) {
            const {
                astronomy,
                atmosphere,
                item,
                cityWeatherLink,
                units,
                wind
            } = weatherInfo;

            currentStatusCard = (
                <CurrentStatusCard
                    atmosphere={ atmosphere }
                    cityWeatherLink={ cityWeatherLink }
                    condition={ item.condition }
                    onUnitToggleChange={ onUnitToggleChange }
                    searchTerm={ searchTerm }
                    units={ units }
                />);
            sunStatusCard = (<SunStatusCard timezone={ timezone } astronomy={ astronomy } />);
            windStatusCard = (
                <WindStatusCard
                    astronomy={ astronomy }
                    timezone={ timezone }
                    units={ units }
                    wind={ wind }
                />
            );
        }

        return (
            <div>
                <Menu
                    searchTerm={ searchTerm }
                    onChange={ onChange }
                    onSubmit={ onSubmit }
                    onHomePageLinkClick={ onHomePageLinkClick }
                    onForecastLinkClick={ onForecastLinkClick }
                />
                <div className="page-container">
                    <ReactCSSTransitionGroup
                        transitionName="loader"
                        transitionEnterTimeout={ 1 }
                        transitionLeaveTimeout={ 1000 }
                    >
                      { isLoading ? <div className="loader-wrapper"></div> : null}
                    </ReactCSSTransitionGroup>
                    { currentStatusCard }
                    { sunStatusCard }
                    { windStatusCard }
                    <div className="external-link-container">
                        <a aria-label="Github link" className="github-link" href={ GITHUB_LINK } target="_blank">
                            <span className="icon-github" aria-hidden="true"></span>
                        </a>
                        <a aria-label="Yahoo weather link"  href={ YAHOO_WEATHER_LINK } target="_blank">
                            <img src={ require('../../assets/img/yahoo-weather.gif') } alt="Yahoo Weather Logo" />
                        </a>
                    </div>
                </div>
            </div>
        );
    }
}
