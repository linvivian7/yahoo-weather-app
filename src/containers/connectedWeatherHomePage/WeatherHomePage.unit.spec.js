import React from 'react';
import WeatherHomePage from './WeatherHomePage';

import * as constants from '../../test/constants';

test('WeatherHomePage snapshot', () => {
    const wrapper = shallow(
        <WeatherHomePage
            searchTerm={ constants.SEARCH_TERM }
            weatherInfo={ constants.WEATHER_INFO }
            timezone={ constants.TIMEZONE }
            isLoading={ false }
        />
    );

    expect(wrapper).toMatchSnapshot();
});
