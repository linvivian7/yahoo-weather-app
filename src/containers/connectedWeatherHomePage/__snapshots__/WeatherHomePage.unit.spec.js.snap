// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHomePage snapshot 1`] = `
<div>
  <Menu
    searchTerm="Taipei, Taiwan"
  />
  <div
    className="page-container"
  >
    <CSSTransitionGroup
      transitionAppear={false}
      transitionEnter={true}
      transitionEnterTimeout={1}
      transitionLeave={true}
      transitionLeaveTimeout={1000}
      transitionName="loader"
    />
    <CurrentStatusCard
      atmosphere={
        Object {
          "humidity": "74",
          "pressure": "994.0",
          "rising": "0",
          "visibility": "16.1",
        }
      }
      cityWeatherLink="https://weather.yahoo.com/country/state/city-2306179/"
      condition={
        Object {
          "code": "27",
          "date": "Fri, 20 Apr 2018 11:00 PM CST",
          "temp": "75",
          "text": "Mostly Cloudy",
        }
      }
      searchTerm="Taipei, Taiwan"
      units={
        Object {
          "distance": "mi",
          "pressure": "in",
          "speed": "mph",
          "temperature": "F",
        }
      }
    />
    <SunStatusCard
      astronomy={
        Object {
          "sunrise": "5:26 am",
          "sunset": "6:19 pm",
        }
      }
      timezone={
        Object {
          "timeZoneId": "America/Los_Angeles",
          "timeZoneName": "Pacific Standard Time",
        }
      }
    />
    <WindStatusCard
      astronomy={
        Object {
          "sunrise": "5:26 am",
          "sunset": "6:19 pm",
        }
      }
      timezone={
        Object {
          "timeZoneId": "America/Los_Angeles",
          "timeZoneName": "Pacific Standard Time",
        }
      }
      units={
        Object {
          "distance": "mi",
          "pressure": "in",
          "speed": "mph",
          "temperature": "F",
        }
      }
      wind={
        Object {
          "chill": "75",
          "direction": "115",
          "speed": "18",
        }
      }
    />
    <div
      className="external-link-container"
    >
      <a
        aria-label="Github link"
        className="github-link"
        href="https://github.com/linvivian7/yahoo-weather-app"
        target="_blank"
      >
        <span
          aria-hidden="true"
          className="icon-github"
        />
      </a>
      <a
        aria-label="Yahoo weather link"
        href="https://weather.yahoo.com/"
        target="_blank"
      >
        <img
          alt="Yahoo Weather Logo"
          src={null}
        />
      </a>
    </div>
  </div>
</div>
`;
